R/cleanifyString.r

```diff
@@ -1,8 +1,18 @@
-cleanifyString <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE, oldEncoding = NULL, ...) {
+cleanifyString <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE,
+                           oldEncoding = NULL, ...) {
+  lapply(c(removeNonAlphaNum, replaceSpecialChars, oldEncoding),
+         checkmate::assert_logical, len = 1)
   UseMethod("cleanifyString")
   }
 
-cleanifyString.character <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE, oldEncoding = NULL, ...) {
+### ----------------------------------------------------------------------------
+
+cleanifyString.character <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE,
+                                     oldEncoding = NULL, ...) {
+  lapply(c(removeNonAlphaNum, replaceSpecialChars, oldEncoding),
+         checkmate::assert_logical, len = 1)
```

Comment:
The first assertion (according to logical) is fine. I think the second consistency check (`assert_character(x)`) could be omitted. Benjamin has changed the function towards S3 methods. This means that `cleanifyString.character()` can only be called if the class has already been confirmed to be character.